services/streammanager/googlepubsub/googlepubsubmanager.go

```diff
@@ -101,6 +106,58 @@ func NewProducer(destination *backendconfig.DestinationT, o common.Opts) (*Googl
 	return &GooglePubSubProducer{client: &PubsubClient{client, topicMap, o}, conf: config.Default}, nil
 }
 
+func (producer *GooglePubSubProducer) publish(ctx context.Context, topic *pubsub.Topic, message *pubsub.Message) (string, error) {
+	result := topic.Publish(ctx, message)
+	serverID, err := result.Get(ctx)
+	if err != nil {
+		return "", err
+	}
+	return serverID, nil
+}
+
+// publishWithRetry publishes a message with retry logic for intermittent authentication errors
+func (producer *GooglePubSubProducer) publishWithRetry(ctx context.Context, topic *pubsub.Topic, message *pubsub.Message) (string, error) {
+	pkgLogger.Infof("[GooglePubSub] Publishing with retry enabled")
+	var serverID string
+
+	operation := func() error {
+		var err error
+		serverID, err = producer.publish(ctx, topic, message)
+		if err == nil {
+			return nil
+		}
+
+		// Only retry on authentication errors
+		st, ok := status.FromError(err)
+		if ok && st.Code() == codes.Unauthenticated {
```

Comment:
I would prefer matching against the exact string, we don't want to retry any authentication error, do we?
```
Request had invalid authentication credentials. Expected OAuth 2 access token, login cookie or other valid authentication credential
```

Reply:
addressed